skbio/sequence/_protein.py

```diff
@@ -82,19 +86,31 @@ class Protein(IUPACSequence):
         has gaps: False
         has degenerates: False
         has non-degenerates: True
+        has stops: False
     -----------------------------
     0 PAW
 
     """
+    __stop_codes = None
 
-    @property
-    def _motifs(self):
-        return _motifs
+    @classproperty
+    def _stop_codes(cls):
+        if cls.__stop_codes is None:
+            stops = cls.stop_chars
+            cls.__stop_codes = np.asarray([ord(s) for s in stops])
+        return cls.__stop_codes
+
+    @classproperty
+    # TODO why doesn't this work?
+    # @overrides(IUPACSequence)
```

Comment:
@ebolyen I get the following error when I uncomment this line:

```
======================================================================
ERROR: Failure: TypeError ('abstractproperty' object is not iterable)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jairideout/.virtualenvs/scikit-bio/lib/python2.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/Users/jairideout/.virtualenvs/scikit-bio/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/Users/jairideout/.virtualenvs/scikit-bio/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/Users/jairideout/dev/scikit-bio/skbio/__init__.py", line 14, in <module>
    import skbio.io  # noqa
  File "/Users/jairideout/dev/scikit-bio/skbio/io/__init__.py", line 313, in <module>
    import_module('skbio.io.clustal')
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/Users/jairideout/dev/scikit-bio/skbio/io/clustal.py", line 111, in <module>
    from skbio.sequence import Sequence
  File "/Users/jairideout/dev/scikit-bio/skbio/sequence/__init__.py", line 285, in <module>
    from ._protein import Protein
  File "/Users/jairideout/dev/scikit-bio/skbio/sequence/_protein.py", line 17, in <module>
    class Protein(IUPACSequence):
  File "/Users/jairideout/dev/scikit-bio/skbio/sequence/_protein.py", line 105, in Protein
    @overrides(IUPACSequence)
  File "/Users/jairideout/dev/scikit-bio/skbio/util/_decorator.py", line 297, in overrider
    method.__doc__ = getattr(interface_class, method.__name__).__doc__
  File "/Users/jairideout/dev/scikit-bio/skbio/util/_decorator.py", line 335, in __get__
    return self.fget.__get__(None, owner)()
  File "/Users/jairideout/dev/scikit-bio/skbio/sequence/_iupac_sequence.py", line 111, in alphabet
    return cls.degenerate_chars | cls.nondegenerate_chars | cls.gap_chars
  File "/Users/jairideout/dev/scikit-bio/skbio/util/_decorator.py", line 335, in __get__
    return self.fget.__get__(None, owner)()
  File "/Users/jairideout/dev/scikit-bio/skbio/sequence/_iupac_sequence.py", line 135, in degenerate_chars
    return set(cls.degenerate_map)
TypeError: 'abstractproperty' object is not iterable
```

Do you have any idea how I can fix this?